nilearn/image/resampling.py

```diff
@@ -455,8 +455,13 @@ def resample_img(img, target_affine=None, target_shape=None,
     target_shape = tuple(target_shape)
 
     if interpolation == 'continuous' and data.dtype.kind == 'i':
-        resampled_data_dtype = np.dtype(
-            data.dtype.name.replace('int', 'float'))
+        # cast unsupported data types to closest support dtype
+        aux = data.dtype.name.replace('int', 'float')
+        aux = aux.replace("ufloat", "float").replace("floatc", "float")
```

Comment:
The rule is even simpler : int64 and float64 lead to float64. Anything else is float32. So it can be written:
```python
aux = "float64" if aux.endswith('64') else "float32"
```